tutorials/haskell/haskell-time-tutorial/tutorial.md

```diff
@@ -0,0 +1,481 @@
+---
+title: Time, patterns and a little more
+published: 2021-11-25
+tags: haskell, pattern-synonyms, property-testing
+libraries: time
+language: haskell
+author: Felix Miño
+author-name: Felix Miño
+github-profile: felixminom
+description: In this tutorial we’re going to explore the use of pattern synonyms in Haskell, later we’ll give a brief overview to the Stack Builders’ contribution to the `time` Haskell library and finally we’ll review how to use the utilities that were introduced in the contribution.
+---
+
+As Stack Builders one of the core values of our company is contributing
+to open source (OSS). We believe that we have the power to change
+people’s lives (or at least make it easier) by pushing the boundaries
+of the software industry and we’re happy to announce that the past month
+we were able to contribute to one of Haskell's core libraries, time.
+
+One of our tech leads, Sebastian Estrella, started with the idea:
+“What if we bring some utilities that Ruby on Rails (RoR) has to Haskell?”.
+For instance, “How can I get the range of days that a quarter has?”
+or “How can I get a month’s boundaries?” were some of the questions
+that came to our mind. In the following example I will show the Haskell
+code needed to get a month's boundaries and its version in RoR.
+
+```haskell
+type MonthFirstDay = Day
```

Comment:
@felixminom since this is a short code snippet I think it is clear that the first element of the tuple corresponds to the first day and the second one to the last day, in that sense, maybe we could get rid of `MonthFirstDay` and `MonthLastDay` type aliases? @jpvillaisaza what do you think?

Reply:
Yes, `monthBoundaries :: Day -> (Day, Day)` sounds better for the tutorial.

Reply:
As I mentioned before I'm going to make both functions to exactly match, so now it will be something like this: `allMonth :: Day -> [Day]`